manticore/core/cpu/abstractcpu.py

```diff
@@ -781,13 +781,15 @@ def emulate(self, insn):
         '''
 
         emu = UnicornEmulator(self)
-        emu.emulate(insn)
-
-
-        # We have been seeing occasional Unicorn issues with it not clearing
-        # the backing unicorn instance. Saw fewer issues with the following
-        # line present.
-        del emu
+        try:
+            emu.emulate(insn)
+        
+        #FIXME(pag): Catch `UcError`, e.g. executing a UD2 instruction.
```

Comment:
Catch the error and log to a `logger` before throwing away?

Reply:
So I wasn't too sure how to get at the right class name, i.e. what imports to make, and also what the right way to propagate the info is. In my mind, `UcError` should be caught and converted into some kind of more manticore-specific exception.